### ISSUE TITLE
Docs for nodegroup and nodegroupV2 show deprecated root block encryption property.

### ISSUE THREAD

**Issue by MitchellGerdisch** [pulumi/pulumi-eks#930]
### What happened?

The docs for the package show this property:
[encryptRootBlockDevice](https://www.pulumi.com/registry/packages/eks/api-docs/nodegroupv2/#encryptrootblockdevice_nodejs)

However, my IDE showed this message when hovering over the property:
```
property) NodeGroupBaseOptions.encryptRootBlockDevice?: Input<boolean> | undefined
Encrypt the root block device of the nodes in the node group.

@deprecated
This option has been deprecated for parameter naming coherence. Use the nodeRootVolumeEncrypted option instead.
```

Although the code accepts the `encryptRootBlockDevice` property, it is ignored.
One must set the undocumented `nodeRootVolumeEncrypted` property for this to work.

### Example

fwiw: https://gist.github.com/MitchellGerdisch/35e2c1ecee3f47dc01bf7f72b16213c8

### Output of `pulumi about`

CLI          
Version      3.89.0
Go Version   go1.21.1
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.42.0
aws         5.31.0
awsx        1.0.6
docker      3.6.1
eks         1.0.4
kubernetes  3.30.2
nodejs      unknown

Host     
OS       darwin
Version  13.5
Arch     x86_64

This project is written in nodejs: executable='/Users/mitch/.nvm/versions/node/v18.0.0/bin/node' version='v18.0.0'

Backend        
Name           pulumi.com

Token type     personal

Dependencies:
NAME            VERSION
@pulumi/aws     5.42.0
@pulumi/awsx    1.0.6
@pulumi/eks     1.0.4
@pulumi/pulumi  3.89.0
@types/node     16.18.59

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by MitchellGerdisch** [comment]
This is related to https://github.com/pulumi/pulumi-eks/issues/926

**Comment by mikhailshilkov** [comment]
We should deprecate this option in V2.

**Comment by thomas11** [comment]
The fix is committed to the v2 branch and will be released with v2.0.
